mobile-app/app/screens/AppNavigator/screens/Dex/hook/SlippageTolerance.ts

```diff
@@ -0,0 +1,30 @@
+import { SlippageTolerancePersistence } from '@api'
+import { useLogger } from '@shared-contexts/NativeLoggingProvider'
+import BigNumber from 'bignumber.js'
+import { useEffect, useState } from 'react'
+
+export interface SlippageTolerance {
+    slippage: BigNumber
+    setSlippage: (val: BigNumber) => Promise<void>
+}
+
+export function useSlippageTolerance (): SlippageTolerance {
+    const logger = useLogger()
+    const [slippage, setSlippage] = useState(new BigNumber(1))
+
+    useEffect(() => {
+        SlippageTolerancePersistence.get().then((slippageValue: BigNumber) => {
+          setSlippage(new BigNumber(slippageValue))
+        }).catch(logger.error)
+    }, [])
+
+    const updateSlippageTolerance = async (slippageVal: BigNumber): Promise<void> => {
+        setSlippage(new BigNumber(slippageVal))
```

Comment:
if your argument is already of type BigNumber, do you still need `new BigNumber()`?